yt_dlp/extractor/mainstreaming.py

```diff
@@ -0,0 +1,222 @@
+            'only_matching': True
+        }, {
+            'url': 'https://859c1818ed614cc5b0047439470927b0.msvdn.net/content/tDoFkZD3T1Lw#',
+            'only_matching': True
+        }
+    ]
+
+    @staticmethod
+    def _extract_urls(webpage):
+        mobj = re.findall(
+            r'<iframe[^>]+?src=["\']?(?P<url>%s)["\']?' % MainStreamingIE._VALID_URL, webpage)
+        if mobj:
+            return [group[0] for group in mobj]
+
+    def _playlist_entries(self, host, playlist_content):
+        for entry in playlist_content:
+            content_id = entry.get('contentID')
+            yield {
+                '_type': 'url',
+                'ie_key': MainStreamingIE.ie_key(),
+                'id': content_id,
+                'duration': int_or_none(traverse_obj(entry, ('duration', 'totalSeconds'))),
+                'title': entry.get('title'),
+                'url': f'https://{host}/embed/{content_id}'
+            }
+
+    @staticmethod
+    def _get_webtools_host(host):
+        if not host.startswith('webtools'):
+            host = 'webtools' + ('-' if not host.startswith('.') else '') + host
+        return host
+
+    def _get_webtools_base_url(self, host):
+        return f'{self.http_scheme()}//{self._get_webtools_host(host)}'
+
+    def _call_api(self, host: str, path: str, item_id: str, query=None, note='Downloading API JSON', fatal=False):
+        # JSON API, does not appear to be documented
+        return self._call_webtools_api(host, '/api/v2/' + path, item_id, query, note, fatal)
+
+    def _call_webtools_api(self, host: str, path: str, item_id: str, query=None, note='Downloading webtools API JSON', fatal=False):
+        # webtools docs: https://webtools.msvdn.net/
+        return self._download_json(
+            urljoin(self._get_webtools_base_url(host), path), item_id, query=query, note=note, fatal=fatal)
+
+    def _real_extract(self, url):
+        host, video_id = self._match_valid_url(url).groups()
+        content_info = try_get(
+            self._call_api(
+                host, f'content/{video_id}', video_id, note='Downloading content info API JSON'), lambda x: x['playerContentInfo'])
+        # Fallback
+        if not content_info:
+            webpage = self._download_webpage(url, video_id)
+            player_config = self._parse_json(
+                self._search_regex(
+                    r'config\s*=\s*({.+?})\s*;', webpage, 'mainstreaming player config',
+                    default='{}', flags=re.DOTALL),
+                video_id, transform_source=js_to_json, fatal=False) or {}
+            content_info = player_config['contentInfo']
+
+        host = content_info.get('host') or host
+        video_id = content_info.get('contentID') or video_id
+        title = content_info.get('title')
+        description = traverse_obj(
+            content_info, 'longDescription', 'shortDescription', expected_type=str, get_all=False)
```

Comment:
```suggestion
        description = traverse_obj(content_info, 'longDescription', 'shortDescription', expected_type=str)
```
get_all isnt applicable here